mdpow/fep.py

```diff
@@ -1349,7 +1350,9 @@ def p_transfer(G1, G2, **kwargs):
         if not hasattr(G, 'stop'):
             G.stop = kwargs.pop('stop', None)
         if not hasattr(G, 'SI'):
-            G.SI = kwargs.pop('SI', False)
+            kwargs['SI'] = kwargs.pop('SI', False)
+        else:
+            kwargs['SI'] = kwargs.pop('SI', G.SI)
```

Comment:
`dict.setdfault()` should do what you want (set the item unless it already exists)
```suggestion
            kwargs.setdefault('SI', False)
        else:
            kwargs.setdefault('SI', G.SI)
```